internal/fsutils/fsutils.go

```diff
@@ -91,3 +92,44 @@ func EvalSymlinks(path string) (string, error) {
 
 	return er.path, er.err
 }
+
+// code from https://github.com/guillermo/doubleglob/blob/main/double_glob.go
+var replaces = regexp.MustCompile(`(\.)|(\*\*/)|(\*)|([^/\*.]+)|(/)`)
+
+func toRegexp(pattern string) string {
+	pat := replaces.ReplaceAllStringFunc(pattern, func(s string) string {
+		switch s {
+		case "/":
+			return "\\/"
+		case ".":
+			return "\\."
+		case "**/":
+			return ".*"
+		case "*":
+			return "[^/]*"
+		default:
+			return s
+		}
+	})
+	return "^" + pat + "$"
+}
+
+func FileNameMatchMask(name, pattern string) bool {
+	regexpPat := regexp.MustCompile(toRegexp(pattern))
+
+	matched := regexpPat.MatchString(name)
+
+	return matched
```

Comment:
Why not just:
```
return regexp.MustCompile(toRegexp(pattern)).MatchString(name)
```
??

In principle, it doesn't matter here, I'm just asking.

Reply:
Just for easier code readability :smile_cat: